src/Controls/src/Core/Compatibility/Handlers/Shell/Android/ShellRenderer.cs

```diff
@@ -345,6 +346,31 @@ void IElementHandler.Invoke(string command, object args)
 
 		void IElementHandler.DisconnectHandler()
 		{
+			if (_disposed)
+				return;
+
+			_disposed = true;
+
+			Element.PropertyChanged -= OnElementPropertyChanged;
+			Element.SizeChanged -= OnElementSizeChanged;
+			((IShellController)Element).RemoveAppearanceObserver(this);
+
+			// This cast is necessary because IShellFlyoutRenderer doesn't implement IDisposable
```

Comment:
Can we include a comment to implement `IDisposable` on net7?

Reply:
This comment was incorrectly copied from the source code. We don't need it to implement IDisposable here. The only point of this check really is if a user has supplied a custom type here then we default back to IDisposable vs IDisconnect